python/dgl/nn/pytorch/conv/dotgatconv.py

```diff
@@ -0,0 +1,93 @@
+"""Torch modules for graph attention networks(GAT)."""
+# pylint: disable= no-member, arguments-differ, invalid-name
+from torch import nn
+
+from .... import function as fn
+from ..softmax import edge_softmax
+from ....utils import expand_as_pair
+
+
+class DotGatConv(nn.Module):
+    r"""Apply dot product version of self attention in GCN.
+
+        .. math::
+            h_i^{(l+1)} = \sum_{j\in \mathcal{N}(i)} \alpha_{i, j} h_j^{(l)}
+
+        where :math:`\alpha_{ij}` is the attention score bewteen node :math:`i` and node :math:`j`:
+
+        .. math::
+            \alpha_{i, j} = \mathrm{softmax_i}(e_{ij}^{l})
+
+            e_{ij}^{l} = ({W_i^{(l)} h_i^{(l)}})^T \cdot {W_j^{(l)} h_j^{(l)}}
+
+        where :math:`W_i` and :math:`W_j` transform node :math:`i`'s and node :math:`j`'s
+        features into the same dimension, so that when compute note features' similarity,
+        we can use dot-product.
+    """
```

Comment:
Please introduce `__init__` function parameters in the docstring by
```
Parameters
----------------
in_feats : int
    your description
out_feats : int
    your description
```

Reply:
Added doc strings about them.